QEfficient/cloud/infer.py

```diff
@@ -30,6 +31,56 @@
 4. Download HF model -> transform -> export -> compile -> execute
 """
 
+def _construct_qpc_dir_name(num_cores: int, batch_size: int, prompt_len: int, ctx_len: int, mos: int, device_group: List[int], mxfp6: bool, mxint8: bool) -> str:
+    """
+    Construct the base directory name for QPC files
+    Parameters:
+    - num_cores (int): Number of cores to use in AI 100.
+    - batch_size (int): Batch size for processing.
+    - prompt_len (int): Length of the prompt.
+    - ctx_len (int): Context length.
+    - mos (int): Maximum out-channel split. 
+    - device_group (List[int]): List of device IDs.
+    - mxfp6 (bool): Flag for enabling MXFP6 precision.
+    - mxint8 (bool): Flag for mixed INT8 precision.
+
+    Returns:
+    - str: The constructed directory name.
+    """
+    qpc_base_dir_name = (
+        f"qpc_{num_cores}cores_{batch_size}BS_{prompt_len}PL_{ctx_len}CL_{mos}MOS_"
+        + f"{len(device_group)}"
+        + "devices"
+        + ("_mxfp6_mxint8" if (mxfp6 and mxint8) else "_mxfp6" if mxfp6 else "_fp16_mxint8" if mxint8 else "_fp16")
+    )
+    return qpc_base_dir_name
+
+def _get_tokenizer(model_name: str, cache_dir: str, hf_token: str = None) -> Tuple[AutoTokenizer, str]:
+    """
+    Login with the Hugging Face token if provided and download the model,
+    Create and return the tokenizer.
+
+    Parameters:
+    - model_name (str): Model name to download.
+    - cache_dir (str): The directory where the model should be cached.
+    - hf_token (Optional[str]): The Hugging Face token for authentication.
+
+    Returns:
+    - AutoTokenizer: The tokenizer for the downloaded model.
+    - str: Downloaded model path
+    """
+    if hf_token is not None:
+        login(hf_token)
+    
+    model_hf_path = hf_download(
```

Comment:
Does it make sense to download entire model checkpoints just to get `tokenizer`?

Reply:
it would be better to move it to a separate method.

Reply:
Hey @quic-rishinr  this loading only tokenizer is being handled in PR 31, please check